texar/tf/modules/embedders/embedder_base.py

```diff
@@ -76,11 +72,12 @@ def _get_dropout_layer(self, hparams, ids_rank=None, dropout_input=None,
             if st == 'element':
                 noise_shape = None
             elif st == 'item':
-                noise_shape = tf.concat([tf.shape(dropout_input)[:ids_rank],
-                                         tf.ones([self._dim_rank], tf.int32)],
-                                        axis=0)
+                assert dropout_input is not None
+                assert ids_rank is not None
+                noise_shape = (dropout_input.get_shape().as_list()[:ids_rank]
```

Comment:
Texar utils has `shape_list` https://texar.readthedocs.io/en/latest/code/utils.html#texar.tf.utils.shape_list

BTW, the doc of `shape_list` is not visualized correctly, could you fix?